### PR TITLE
Flag details test

### PR DESCRIPTION
**Pull request by kallehauge**

### CODE REVIEW

**Comment by kasperg** [comment]
@kallehauge: This should be ready for review now.

File: web/modules/custom/dbcdk_community/tests/src/Unit/Plugin/Block/BlockTestBase.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+   *   The id of the row to check.
+   * @param string|null $message
+   *   The message to display if the assertion fails.
+   */
+  protected function assertDetailsRowNotEmpty(array $table, $row_id, $message = NULL) {
+    $this->assertNotEmpty($table['#rows'][$row_id][1], $message);
+  }
+
+  /**
+   * Assert that an entry is a details table is a link to a specific url.
```
Comment (kallehauge) [inline web/modules/custom/dbcdk_community/tests/src/Unit/Plugin/Block/BlockTestBase.php]: This comment sounds a bit funny around here:

> "... an entry is a details table is a link to..."

File: web/modules/custom/dbcdk_community/tests/src/Unit/Plugin/Block/BlockTestBase.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+   *   The id of the row to check.
+   * @param string|NULL $message
+   *   The message to display if the assertion fails.
+   */
+  protected function assertDetailsRowEquals($expected, array $table, $row_id, $message = NULL) {
+    $this->assertEquals($expected, $table['#rows'][$row_id][1], $message);
+  }
+
+  /**
+   * Assert that an entry is a details table is not empty.
```
Comment (kallehauge) [inline web/modules/custom/dbcdk_community/tests/src/Unit/Plugin/Block/BlockTestBase.php]: This comment sounds a bit funny around here:

> "... an entry is a details table is not empty."

File: web/modules/custom/dbcdk_community/tests/src/Unit/Plugin/Block/BlockTestBase.php [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+   */
+  protected function assertDetailsRowLinkUrl($expected_url, $table, $row_id, $message = NULL) {
+    $this->assertInstanceOf('\Drupal\Core\Link', $table['#rows'][$row_id][1]);
+    /* @var \Drupal\Core\Link $actual_link */
+    $actual_link = $table['#rows'][$row_id][1];
+    $this->assertEquals($expected_url, $actual_link->getUrl()->getUri());
+  }
+
+  /**
+   * Assert that an entry is a details table is a link with a specific text.
```
Comment (kallehauge) [inline web/modules/custom/dbcdk_community/tests/src/Unit/Plugin/Block/BlockTestBase.php]: This comment sounds a bit funny around here:

> "... an entry is a details table is a link..."

File: web/modules/custom/dbcdk_community/tests/src/Unit/Plugin/Block/FlaggedContentDetailsBlockTest.php [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    $flagged_content_details = $this->newFlaggedContentDetailsBlock();
+
+    $result = $flagged_content_details->build();
+
+    // When an exception is thrown there should be no details.
+    $this->assertNoDetails($result['table']);
+  }
+
+  /**
+   * Test handling when trying to retrieve an flagged content with invalid id.
```
Comment (kallehauge) [inline web/modules/custom/dbcdk_community/tests/src/Unit/Plugin/Block/FlaggedContentDetailsBlockTest.php]: Typo: `an flagged content` => `a flagged content`

**Comment by kallehauge** [comment]
Scrutinizer found some easy things to correct like type-hinting arrays etc.
Besides that: It looks great! :+1: I only found some small things in comments (typo's etc).

File: web/modules/custom/dbcdk_community/tests/src/Unit/Plugin/Block/BlockTestBase.php [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+   * @param string|int $row_id
+   *   The id of the row to check.
+   * @param string|null $message
+   *   The message to display if the assertion fails.
+   */
+  protected function assertDetailsRowLinkUrl($expected_url, $table, $row_id, $message = NULL) {
+    $this->assertInstanceOf('\Drupal\Core\Link', $table['#rows'][$row_id][1]);
+    /* @var \Drupal\Core\Link $actual_link */
+    $actual_link = $table['#rows'][$row_id][1];
+    $this->assertEquals($expected_url, $actual_link->getUrl()->getUri());
```
Comment (kallehauge) [inline web/modules/custom/dbcdk_community/tests/src/Unit/Plugin/Block/BlockTestBase.php]: Scrutinizer found this: use `$message` in `$this->assertEquals()`

**Comment by kasperg** [comment]
This can be merged now.

**Pull request merged**